compiler/rustc_passes/src/dead.rs

```diff
@@ -505,6 +506,13 @@ fn find_live<'tcx>(
     symbol_visitor.live_symbols
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+enum ExtraNote {
+    /// Use this to provide some examples in the diagnostic of potential other purposes for a value
+    /// or field that is dead code
+    OtherPurposeExamples,
```

Comment:
Happy to hear if you have suggestions for better names for this variant and the enum itself.

Reply:
Well, its not a public enum, so I think the names can stay as they are. (I'd spend more time worrying about the names if they were going to get exposed outside this module.)

Reply:
If we ever come to a point where there are other variants here, I assume this enum will change so that the variant indicates the *context* where the warning is coming from. E.g. saying "this came from an unused field" vs "this came from an unused impl", etc.)